src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,427 @@
+            # k1
+            integrator.f(integrator.du, integrator.u, prob.p, integrator.t)
+            @threaded for i in eachindex(integrator.du)
+                integrator.k1[i] = integrator.du[i] * integrator.dt
+            end
+
+            # Construct current state
+            @threaded for i in eachindex(integrator.du)
+                integrator.u_tmp[i] = integrator.u[i] + alg.c[2] * integrator.k1[i]
+            end
+            # k2
+            integrator.f(integrator.du, integrator.u_tmp, prob.p,
+                         integrator.t + alg.c[2] * integrator.dt)
+
+            @threaded for i in eachindex(integrator.du)
+                integrator.k_higher[i] = integrator.du[i] * integrator.dt
+            end
+
+            if alg.num_stages == 3
+                @threaded for i in eachindex(integrator.du)
+                    integrator.k_S1[i] = integrator.k_higher[i]
+                end
+            end
+
+            # Higher stages
+            for stage in 3:(alg.num_stages - 1)
+                # Construct current state
+                @threaded for i in eachindex(integrator.du)
+                    integrator.u_tmp[i] = integrator.u[i] +
+                                          alg.a_matrix[stage - 2, 1] *
+                                          integrator.k1[i] +
+                                          alg.a_matrix[stage - 2, 2] *
+                                          integrator.k_higher[i]
+                end
+
+                integrator.f(integrator.du, integrator.u_tmp, prob.p,
+                             integrator.t + alg.c[stage] * integrator.dt)
+
+                @threaded for i in eachindex(integrator.du)
+                    integrator.k_higher[i] = integrator.du[i] * integrator.dt
+                end
+            end
+
+            @threaded for i in eachindex(integrator.du)
+                integrator.k_S1[i] = integrator.k_higher[i]
+            end
+
+            # Last stage
+            @threaded for i in eachindex(integrator.du)
+                integrator.u_tmp[i] = integrator.u[i] +
+                                      alg.a_matrix[alg.num_stages - 2, 1] *
+                                      integrator.k1[i] +
+                                      alg.a_matrix[alg.num_stages - 2, 2] *
+                                      integrator.k_higher[i]
+            end
+
+            integrator.f(integrator.du, integrator.u_tmp, prob.p,
+                         integrator.t + alg.c[alg.num_stages] * integrator.dt)
+            @threaded for i in eachindex(integrator.du)
+                integrator.k_higher[i] = integrator.du[i] * integrator.dt
+            end
+
+            @threaded for i in eachindex(integrator.u)
+                # "Own" PairedExplicitRK based on SSPRK33
+                integrator.u[i] += (integrator.k1[i] + integrator.k_S1[i] +
+                                    4.0 * integrator.k_higher[i]) / 6.0
```

Comment:
Potentially we could also make this more efficient by not writing back again to `k_higher` in line 387, but directly use `integrator.du * integrator.dt` in line 393. 
Can you try this as well @warisa-r ?

Reply:
Alright! So it doesn't make any difference in runtime but if you deem this more elegant I can push the change.

Reply:
Yeah why not, less code is in most cases a good thing.